custom/enikshay/integrations/nikshay/repeater_generator.py

```diff
@@ -0,0 +1,273 @@
+        # "atbsource": "G",
+        # "atbregimen": "Manish-Regim",
+        # "atbyr": "2015",
+        # "dcpulmunory": "P",
+        # "IP_Address": "mk-ip-address",
+        # "IP_From": "127.0.0.1",
+        person_case = get_person_case_from_episode(episode_case.domain, episode_case.get_id)
+
+        episode_case_properties = episode_case.dynamic_case_properties()
+        person_case_properties = person_case.dynamic_case_properties()
+        properties_dict = {
+            "regBy": CouchUser.get(person_case.opened_by).human_friendly_name,
+            "Local_ID": person_case.get_id,
+            "Source": ENIKSHAY_ID,
+            "dotcenter": "NA",
+            "IP_From": "127.0.0.1",
+        }
+        try:
+            properties_dict.update(_get_person_case_properties(person_case, person_case_properties))
+        except NikshayLocationNotFound as e:
+            _save_error_message(person_case.domain, person_case.case_id, e)
+        properties_dict.update(_get_episode_case_properties(episode_case_properties))
+        return json.dumps(properties_dict)
+
+    def handle_success(self, response, payload_doc, repeat_record):
+        # A success would be getting a nikshay_id for the patient
+        # without it this would actually be a failure
+        try:
+            nikshay_id = _get_nikshay_id_from_response(response)
+            update_case(
+                payload_doc.domain,
+                payload_doc.case_id,
+                {
+                    "nikshay_registered": "true",
+                    "nikshay_id": nikshay_id,
+                    "nikshay_error": "",
+                },
+                external_id=nikshay_id,
+            )
+        except NikshayResponseException as e:
+            _save_error_message(payload_doc.domain, payload_doc.case_id, e.message)
+
+    def handle_failure(self, response, payload_doc, repeat_record):
+        if response.status_code == 409:  # Conflict
+            update_case(
+                payload_doc.domain,
+                payload_doc.case_id,
+                {
+                    "nikshay_registered": "true",
+                    "nikshay_error": "duplicate",
+                },
+            )
+        else:
+            _save_error_message(payload_doc.domain, payload_doc.case_id, unicode(response.json()))
+
+    def handle_exception(self, exception, repeat_record):
+        if isinstance(exception, RequestConnectionError):
+            _save_error_message(repeat_record.domain, repeat_record.payload_id, unicode(exception))
+
+
+def _get_nikshay_id_from_response(response):
+    try:
+        response_json = response.json()
+    except ValueError:
+        raise NikshayResponseException("Invalid JSON received")
+
+    try:
+        message = response_json['Nikshay_Message']
+        if message == "Success":
+            results = response_json['Results']
+            nikshay_ids = [result['Fieldvalue'] for result in results if result['FieldName'] == 'NikshayId']
```

Comment:
no (hence the check on line 128). I am however, expecting there to be more than one "result" key/value pair, and this was the easiest way for me to extract the one which had "NikshayId" as the key value.